compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -177,10 +177,17 @@ pub enum Operand<Tag: Provenance = AllocId> {
 pub struct OpTy<'tcx, Tag: Provenance = AllocId> {
     op: Operand<Tag>, // Keep this private; it helps enforce invariants.
     pub layout: TyAndLayout<'tcx>,
+    /// rustc does not have a proper way to represent the type of a field of a `repr(packed)` struct:
+    /// it needs to have a different alignment than the field type would usually have.
+    /// So we represent this here with a separate field that "overwrites" `layout.align`.
+    /// This means `layout.align` should never be used for an `OpTy`!
+    /// `None` means "alignment does not matter since this is a by-value operand"
+    /// (`Operand::Immediate`).
+    pub align: Option<Align>,
```

Comment:
So afaict this field is never used, and after lots of going through this PRs diff, I came to the conclusion it will only be used to check reads from `Operand::Indirect`, and only in miri.

Please document the miri part, the other part is already documented after all.

Reply:
Yeah currently this field cannot cause UB I think. But I want to relax the rules for `*ptr` and once we do, this *can* cause UB again.